src/views/main/components/make-quiz-drawer-steps.tsx

```diff
@@ -0,0 +1,134 @@
+import { FakeSelectTrigger } from '@/shared/components/fake-select-trigger'
+import { Drawer, DrawerContent, DrawerTrigger } from '@/shared/components/ui/drawer'
+import SelectDrawerContent from './select-drawer-content'
+import SelectCheckItems from './select-check-items'
+import { ReturnUseCheckList, SelectCategoryItem, SelectDocumentItem } from '@/types/quiz'
+import { CategoryDTO } from '@/actions/types/dto/category.dto'
+import { useState } from 'react'
+
+// MakeQuizDrawer 내부에서 사용되는 컴포넌트들
+interface Props {
+  categories: CategoryDTO[]
+  step: 'folder' | 'document'
+  setStep: (step: 'folder' | 'document') => void
+  folder: ReturnUseCheckList<SelectCategoryItem>
+  doc: ReturnUseCheckList<SelectDocumentItem>
```

Comment:
따로 타입을 작성해주는 것도 좋지만 아래와 같이 타입스크립트에서 제공하는 타입을 활용하는 것도 좋을 것 같아요!

```suggestion
  folder: ReturnType<typeof useCheckList<SelectCategoryItem>>
  doc: ReturnType<typeof useCheckList<SelectDocumentItem>>
```